velox/exec/TableWriter.cpp

```diff
@@ -249,6 +253,42 @@ void TableWriter::close() {
   }
 }
 
+void TableWriter::setConnectorOrWriterMemoryReclaimer(
+    memory::MemoryPool* pool) {
+  VELOX_CHECK_NOT_NULL(pool);
+  pool->setReclaimer(
+      TableWriter::MemoryReclaimer::create(operatorCtx_->driverCtx(), this));
+}
+
+std::unique_ptr<memory::MemoryReclaimer> TableWriter::MemoryReclaimer::create(
+    DriverCtx* driverCtx,
+    Operator* op) {
+  return std::unique_ptr<memory::MemoryReclaimer>(
+      new TableWriter::MemoryReclaimer(
+          driverCtx->driver->shared_from_this(), op));
```

Comment:
maybe
return std::make_unique<<TableWriter::MemoryReclaimer>>(driverCtx->driver->shared_from_this(), op);

Reply:
We can't MemoryReclaimer is protected member